### PR TITLE
✨ What in animation tarnation

### PR DESCRIPTION
**Pull request by henryboldi**
Have you ever wondered what a world would be like with animated page transitions in Tally? You and me are alike. It's as if a tranquil sunset washes over each of your clicks in the UI.

To make this possible we've rejiggered some things,
- `CorePage` is wrapped in `Popup.tsx` rather than again and again in pages
- `TopMenu` acts independently from core page
- Added the `react-transition-group` package (5,438,627 weekly downloads)
- Created a config file for routes in `utils/routes.tsx`
- The bulk of the animation logic is handled in `pageTransition.ts`

These animations feature a couple details,
- Top bar stays still when appropriate
- Slides in different directions based on a couple factors,
        - What the clicked tab bar icon's location is relative to the current tab
        - Back or forward in history
        

https://user-images.githubusercontent.com/1918798/147542986-bc07f0bd-7bfc-4d9c-b702-deefce488192.mp4

### CODE REVIEW

File: ui/components/TabBar/TabBarIcon.tsx [hunk 1]
```diff
@@ -1,35 +1,79 @@
...
+          .active span {
+            opacity: 1;
+          }
+
+          .tab_bar_icon_wrap:hover,
+          .active {
+            transform: translateY(-8px);
+          }
+          .active span {
+            color: #d08e39;
```
Comment (wojciech-turek) [inline ui/components/TabBar/TabBarIcon.tsx]: Is this custom or can we use a variable?

File: ui/pages/DAppConnectRequest.tsx [hunk 2]
```diff
@@ -110,65 +109,65 @@ export default function DAppConnectRequest(): ReactElement {
...
           display: flex;
           flex-direction: column;
           height: 100vh;
+          margin: 0 auto;
         }
         .page {
           background-color: var(--green-95);
           height: 100vh;
           width: 100vw;
+          z-index: 1000;
```
Comment (wojciech-turek) [inline ui/pages/DAppConnectRequest.tsx:170]: Its easy to get lost in the z-indexes, we should make a proper structure for it, like X is for modals so i know I need X+1 for something absolute to show over a modal.

File: ui/pages/Earn.tsx [hunk 3]
```diff
@@ -80,26 +79,24 @@ function EarnCard() {
 export default function Earn(): ReactElement {
   return (
     <>
-      <CorePage>
```
Comment (wojciech-turek) [inline ui/pages/Earn.tsx:83]: Seems we are ditching `CorePage` use in a lot of places, was its only purpose before to display the topBar and apply some container styles?

File: ui/pages/Popup.tsx [hunk 4]
```diff
@@ -50,66 +51,83 @@ function transformLocation(inputLocation: Location): Location {
...
+                                  hasTabBar={hasTabBar}
+                                  hasTopBar={hasTopBar}
+                                >
+                                  <Component location={transformedLocation} />
+                                </CorePage>
+                              </Route>
+                            )
+                          }
+                        )}
+                      </Switch>
```
Comment (wojciech-turek) [inline ui/pages/Popup.tsx:119]: Me gusta <3

**Review by wojciech-turek — COMMENTED** [review]
Loving this PR, left some small notes and questions, nothing blocking.

Fantastic job!

File: ui/pages/Earn.tsx (hunk 3 above)
Comment (henryboldi) [reply ui/pages/Earn.tsx:83]: Pretty much. `CorePage` also contains the `SnackBar` component and community edition label. Previously it also handled displaying a variety of slide up menus, but now those have been moved to `TopMenu`. So `CorePage` is a lot less chunky and, making use of `routes.tsx`, is wrapping all of the pages via `Popup.tsx`

File: ui/pages/DAppConnectRequest.tsx (hunk 2 above)
Comment (henryboldi) [reply ui/pages/DAppConnectRequest.tsx:170]: Good point! I've got the bad habit of just putting a high number 😅

**Review by mhluongo — APPROVED** [review]
Approving based on @wojciech-turek's comments offline. Polish is nice 💅

**Pull request merged**